src/gluonts/dataset/split/splitter.py

```diff
@@ -198,61 +183,69 @@ def _train_slice(self, item: TimeSeriesSlice) -> TimeSeriesSlice:
 
     @abstractmethod
     def _test_slice(
-        self, item: TimeSeriesSlice, offset: int = 0
-    ) -> TimeSeriesSlice:
+        self, item: TimeSeriesSlice, prediction_length: int, offset: int = 0
+    ) -> Tuple[TimeSeriesSlice, TimeSeriesSlice]:
         pass
 
-    def _trim_history(self, item: TimeSeriesSlice) -> TimeSeriesSlice:
-        if getattr(self, "max_history") is not None:
-            return item[-getattr(self, "max_history") :]
+    def _trim_history(
+        self, item: TimeSeriesSlice, label_len: int, max_history
+    ) -> TimeSeriesSlice:
+        if max_history is not None:
+            return item[-max_history + label_len :]
         else:
             return item
 
-    def split(self, items: List[DataEntry]) -> TrainTestSplit:
-        split = TrainTestSplit()
+    def split(self, items: List[DataEntry]):
```

Comment:
I think we can rename this and fix its type

```suggestion
    def split(self, dataset: Dataset):
```